tests/kernel/testLinearizer.cpp

```diff
@@ -0,0 +1,349 @@
+     *
+     * @param aPoint a point
+     * @param lowerBound lower bound of the image domain.
+     * @param extent extension of the image domain.
+     *
+     * @return the index
+     */
+    static Size apply( const Point & aPoint, const Point & lowerBound,
+        const Point & extent )
+    {
+      Size pos = aPoint[ 0 ] - lowerBound[ 0 ] ;
+      Size multiplier = 1;
+      for (typename Domain::Dimension k = 1 ; k < dimension ; ++k)
+      {
+        multiplier *= extent[ k-1  ];
+        pos += multiplier * ( aPoint[ k ] - lowerBound[ k ] );
+      }
+      return pos;
+    }
+  };
+
+  /**
+   * Specialization of the linearizer class for dimension 1.
+   *
+   */
+  template < typename Domain >
+  struct linearizer< Domain, 1 >
+  {
+    typedef typename Domain::Point Point;
+    typedef typename Domain::Size Size;
+
+    static Size apply( const Point & aPoint,
+        const Point & lowerBound,
+        const Point & /*extent*/ )
+    {
+      return aPoint[ 0 ] - lowerBound[ 0 ];
+    }
+  };
+
+  /**
+   * Specialization of the linearizer class for dimension 2.
+   *
+   */
+  template < typename Domain >
+  struct linearizer< Domain, 2 >
+  {
+    typedef typename Domain::Point Point;
+    typedef typename Domain::Size Size;
+
+    static Size apply( const Point & aPoint,
+        const Point & lowerBound,
+        const Point & extent )
+    {
+      return ( aPoint[ 0 ] - lowerBound[ 0 ] ) + extent[ 0 ] *
+	(aPoint[ 1 ] - lowerBound[ 1 ] );
+    }
+  };
+
+  /**
+   * Specialization of the linearizer class for dimension 3.
+   *
+   */
+  template < typename Domain >
+  struct linearizer< Domain, 3 >
+  {
+    typedef typename Domain::Point Point;
+    typedef typename Domain::Size Size;
+
+    static Size apply( const Point & aPoint,
+        const Point & lowerBound,
+        const Point & extent )
+    {
+      Size res = aPoint[ 0 ] - lowerBound[ 0 ];
+      Size multiplier = extent[ 0 ];
+      res += multiplier * ( aPoint[ 1 ] - lowerBound[ 1 ] );
+      multiplier *= extent[ 1 ];
+      res += multiplier * ( aPoint[ 2 ] - lowerBound[ 2 ] );
+      return res;
+    }
+  };
+}
+
+
+/// Converter between col-major and row-major storage order.
+template < typename StorageOrder >
+struct PointConverter;
+
+template <>
+struct PointConverter<ColMajorStorage>
+{
+  template < typename TPoint >
+  static inline
+  TPoint apply( TPoint const& aPoint )
+    {
+      return aPoint;
+    }
+};
+
+template <>
+struct PointConverter<RowMajorStorage>
+{
+  template < typename TPoint >
+  static inline
+  TPoint apply( TPoint const& aPoint )
+    {
+      TPoint result;
+      for ( std::size_t i = 0 ; i < TPoint::dimension ; ++i )
+        result[i] = aPoint[ TPoint::dimension - i - 1 ];
+
+      return result;
+    }
+};
+
+#define TEST_LINEARIZER( N , ORDER ) \
+TEST_CASE( "Testing Linearizer in dimension " #N " with " #ORDER, "[test][dim" #N "][" #ORDER "]" )\
```

Comment:
:smile:  nice macro based test case generator ;)
unfortunately, there is no such thing in catch yet https://github.com/philsquared/Catch/issues/358

or https://github.com/philsquared/Catch/issues/46